### PR TITLE
IndexService independent from participant-state

### PR DESCRIPTION
**Pull request by gerolf-da**
The index service serves as a "bridge" between the data on the ledger and the ledger api users. That is also the reason why the index service API is defined in terms of the types in `ledger-api-domain`. It is the responsibility of the index service implementations to return compatible types and adhere to the expected semantics, so we can actually reuse index service implementations in the ledger api servers.

### Pull Request Checklist

- [ ] Read and understand the [contribution guidelines](https://github.com/digital-asset/daml/blob/master/CONTRIBUTING.md)
- [ ] Include appropriate tests
- [ ] Set a descriptive title and thorough description
- [ ] Add a reference to the [issue this PR will solve](https://github.com/digital-asset/daml/issues), if appropriate
- [ ] Add a line to the [release notes](https://github.com/digital-asset/daml/blob/master/docs/source/support/release-notes.rst), if appropriate

NOTE: CI is not automatically run on non-members pull-requests for security
reasons. The reviewer will have to comment with `/AzurePipelines run` to
trigger the build.

### CODE REVIEW

File: ledger/participant-state-index/src/main/scala/com/daml/ledger/participant/state/index/v2/package.scala [hunk 1]
```diff
@@ -0,0 +1,159 @@
...
+      events: List[AcsUpdateEvent]
+  )
+
+  sealed trait AcsUpdateEvent extends Product with Serializable
+
+  object AcsUpdateEvent {
+
+    final case class Create(
+        transactionId: TransactionId,
+        nodeId: NodeId,
```
Comment (gaboraranyossy-da) [inline ledger/participant-state-index/src/main/scala/com/daml/ledger/participant/state/index/v2/package.scala]: do we actually need this? it looks like we don't need the eventId at all here: https://github.com/digital-asset/daml/pull/1314/files#diff-187da506c796f10dc0d43c95b946dd3fR37

File: ledger/participant-state-index/src/main/scala/com/daml/ledger/participant/state/index/v2/CompletionsService.scala [hunk 2]
```diff
@@ -0,0 +1,21 @@
...
+import com.digitalasset.daml.lf.data.Ref
+import com.digitalasset.ledger.api.domain.{ApplicationId, LedgerOffset}
+
+/**
+  * Serves as a backend to implement
+  * [[com.digitalasset.ledger.api.v1.command_completion_service.CommandCompletionServiceGrpc.CommandCompletionService]]
+  **/
+trait CompletionsService {
+  def getCompletions(
+      beginAfter: Option[LedgerOffset],
```
Comment (gerolf-da) [inline ledger/participant-state-index/src/main/scala/com/daml/ledger/participant/state/index/v2/CompletionsService.scala]: should be `beginAt`.

File: ledger/participant-state-index/src/main/scala/com/daml/ledger/participant/state/index/v2/package.scala (hunk 1 above)
Comment (gerolf-da) [reply ledger/participant-state-index/src/main/scala/com/daml/ledger/participant/state/index/v2/package.scala]: Discussed with @dajmaki: Needs to be an opaque `String` that actually comes from the `ReadService`.

File: ledger/participant-state-index/src/main/scala/com/daml/ledger/participant/state/index/v2/package.scala [hunk 3]
```diff
@@ -0,0 +1,159 @@
...
+        contractId: Value.AbsoluteContractId,
+        templateId: Ref.Identifier,
+        argument: Value.VersionedValue[Value.AbsoluteContractId],
+        // TODO(JM,SM): understand witnessing parties
+        stakeholders: List[Ref.Party],
+    ) extends AcsUpdateEvent
+
+    final case class Archive(
+        transactionId: TransactionId,
+        nodeId: NodeId,
```
Comment (gerolf-da) [inline ledger/participant-state-index/src/main/scala/com/daml/ledger/participant/state/index/v2/package.scala]: see https://github.com/digital-asset/daml/pull/1313/files#r286489644

File: ledger/participant-state-index/src/main/scala/com/daml/ledger/participant/state/index/v2/package.scala [hunk 4]
```diff
@@ -0,0 +1,159 @@
...
+    *
+    * @param submitterInfo: information about the original submission of the transaction. Is [[None]]
+    *   if the participant node does not host the submitter.
+    *
+    * @param blindingInfo: Information for divulgence and disclosure.
+    *
+    */
+  final case class Transaction(
+      transactionData: GenTransaction.WithTxValue[NodeId, AbsoluteContractId],
+      blindingInfo: BlindingInfo,
```
Comment (gerolf-da) [inline ledger/participant-state-index/src/main/scala/com/daml/ledger/participant/state/index/v2/package.scala]: Should not be needed. The filtering that needs this should happen within the index service.

**Comment by gerolf-da** [comment]
* I changed the usage of `NodeId` to a new tagged type `AbsoluteNodeId` (better name suggestions are welcome, but we can change later as well).
* `bindlingInfo` is now removed from transaction. all filtering should happen within the transaction service.

**Pull request merged**